pkg/indexgateway/client.go

```diff
@@ -355,48 +352,11 @@ func (s *GatewayClient) GetShards(
 			return errCt <= maxErrs
 		},
 	); err != nil {
-		if isUnimplementedCallError(err) {
```

Comment:
No (the OSS repo is independent from our internal deployments at Grafana Labs). This is because the new `bounded` shards implementation is an intentional choice that needs to be turned on (default=power_of_two), so I removed the fallback for simplicity's sake.